pkg/test/runner/runner.go

```diff
@@ -206,7 +206,17 @@ func (r *Runner) runFnRender() error {
 	// run function
 	kptArgs := []string{"fn", "render", pkgPath}
 	for i := 0; i < r.testCase.Config.RunCount(); i++ {
-		output, fnErr := runCommand("", r.kptBin, kptArgs)
+		output, stderr, fnErr := runCommand("", r.kptBin, kptArgs)
+		if fnErr != nil {
+			if r.testCase.Config.ExitCode != 0 {
+				expected := r.testCase.Config.StdErr
+				if !strings.Contains(stderr, expected) {
```

Comment:
we should do this comparison only if expectedStderr is != "". Also can we rename `expected` to `expectedStdErr` ?